### ISSUE TITLE
Symlinking a file into a tag directory should tag it

### ISSUE THREAD

**Issue by Similacrest** [oniony/TMSU#69]
```
% ln -s file ~/tmsuroot/tags/test
ln: failed to create symbolic link '/home/sharedvi/tmsuroot/tags/test/file': Function not implemented

```

What I want:

```
% ln -s file ~/tmsuroot/tags/test
% tmsu files test
./file
```

Would simplify GUI tagging, for example.

**Comment by oniony** [comment]
I've looked at this before. The problem is you would have to give it a
unique name. TMSU uses identifiers within the VFS directories to ensure
uniqueness, so that if you tag two files with the same name with the same
tag, they can both be shown in the tag directory.

Now if you're symlinking, you're going to have to somehow know the
identifier up front. I've tried, in the past, allowing symlinking without
the identifier and have TMSU assign it but it creates errors if TMSU
changes the name during the symlinking operation.

The only other option i can think of would be for TMSU to drop the
identifiers but that would cause errors as soon as you have a duplicate
filename.

If you have any suggestions on how to handle this then I'd be happy to hear
them.

**Comment by oniony** [comment]
See also [original post](https://www.reddit.com/r/archlinux/comments/4g7p3o/what_application_do_you_always_install_on_your/d2g9sf1).

On reflection maybe the option of specifying a zero is workable. I think I'll have another stab at this and see how it flies.

**Comment by 0ion9** [comment]
@oniony : 
You write in point 2 of 'original post' link:

> Either way, the meaning of zero will change over time (as new files are added) which could potentially break anything that is relying upon the filename (e.g. a graphical file manager that's been left open).

I'm guessing you reconsidered that evaluation? Personally -- assuming I understand the scenario correctly (link is created with number zero and is renamed to first available nonzero ID at TMSU's next opportunity) -- I would say that is the file manager's responsibility to cope with, as renaming a symlink is a pretty ordinary operation.

**Comment by oniony** [comment]
>assuming I understand the scenario correctly (link is created with number zero and is renamed to first available nonzero ID at TMSU's next opportunity) 

Yes, but if it happens immediately, this can confuse the file manager that created the file when it checks that the link was created successfully (by using 'stat'). I hit this problem when I looked at this issue previously: I used a graphical program to create the link, TMSU assigned it an ID but then the program attempted to 'stat' the file with the zero identifier and TMSU told it the file did not exist. I did consider adding a grace period of a second or so after each file creation, or even until the next file is created, but then my complexity alarm went off and I left it for another day.

**Comment by oniony** [comment]
Anyhow, I'll give this another shot soon: I'm sure I could get the VFS to respond affirmatively with the zero ID name for a second or two after creation, which would allow the link creation to succeed.
